core/ecschema-editing/src/test/Validation/SchemaComparison.test.ts

```diff
@@ -0,0 +1,556 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { Schema, SchemaContext } from "@itwin/ecschema-metadata";
+import { AnyDiagnostic, ISchemaChanges, ISchemaCompareReporter, SchemaChanges, SchemaCompareCodes, SchemaComparer } from "../../ecschema-editing";
+import { expect } from "chai";
+
+class TestSchemaCompareReporter implements ISchemaCompareReporter {
+  public changes: SchemaChanges[] = [];
+  public report(schemaChanges: ISchemaChanges): void {
+    this.changes.push(schemaChanges as SchemaChanges);
+  }
+}
+
+function findDiagnostic(diagnostics: AnyDiagnostic[], code: string,fullNameA: string, fullNameB: string, propertyType?: string) {
+  let found = false;
+
+  diagnostics.find((anyDiagnostic) => {
+    switch (code) {
+      case SchemaCompareCodes.BaseClassDelta : {
+        if (anyDiagnostic.code === code &&
+          anyDiagnostic.messageArgs?.at(0).fullName === fullNameA &&
+          anyDiagnostic.messageArgs?.at(1).fullName === fullNameB) {
+          found = true;
+        }
+        break;
+      }
+      case SchemaCompareCodes.PropertyDelta : {
+        if (anyDiagnostic.code === code &&
+          anyDiagnostic.messageArgs?.at(0) === propertyType &&
+          anyDiagnostic.messageArgs?.at(1) === fullNameA &&
+          anyDiagnostic.messageArgs?.at(2) === fullNameB) {
+          found = true;
+        }
+        break;
+      }
+    }
+
+  });
+
+  return found;
+}
+
+describe("Schema comparison tests to filter out cases", () => {
+  let reporter: TestSchemaCompareReporter;
+  let contextA: SchemaContext;
+  let contextB: SchemaContext;
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
you are using this variable, is this suppression really needed?

Reply:
No it's not needed.